### ISSUE TITLE
Nest.GeoLocation lat and lon JProperty Attributes are ignored

### ISSUE THREAD

**Issue by hvanzelst** [elastic/elasticsearch-net#3669]
**NEST/Elasticsearch.Net version**: 6.4.0

**Elasticsearch version**:

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:
Looks like the GeoLocation class properties Latitude and Lontitude in Nest 6.4.0 the JsonProperty("lon") is ignored. Failing the call to elastic, as it requires "lon" and "lat"

**Comment by russcam** [comment]
Would you be able to provide a small succinct example that reproduces what you're seeing?

**Comment by russcam** [comment]
We’re closing this issue due to inactivity; if you’re still in need of resolution, feel free to re-open with the additional details and we'd be glad to help you out 👍

**Comment by Philo** [comment]
I am experiencing this same issue with NEST v6.8.2, see test case below:

```c#
        [Fact]
        public void GeoLocationDoesNotHonourJsonPropertyAttributes()
        {
            //From GeoLocation.cs
            //[JsonProperty("lat")]
            //public double Latitude
            //{
            //    get;
            //}

            //[JsonProperty("lon")]
            //public double Longitude
            //{
            //    get;
            //}

            var p = new Nest.GeoLocation(1.23, 5.67);
            var expectedJson = "{\"lat\":1.23,\"lon\":5.67}";
            var actualJson = "{\"Latitude\":1.23,\"Longitude\":5.67}";

            var json = JsonConvert.SerializeObject(p);
            
            json.Should().NotBe(expectedJson);
            
            json.Should().Be(actualJson);
        }
```

**Comment by Philo** [comment]
Hi @russcam, would it be possible for someone to review and comment on this issue and re-open if appropriate.

Thanks

**Comment by russcam** [comment]
@Philo the test case isn't representative, because the client does not _directly_ use Json.NET's `JsonConvert` type to serialize anything. 

The 6.x client does use an internalized version of Json.NET which can be considered to have similar behaviour to Json.NET, _but_ **only** for the internalized Json.NET types i.e. for all intents and purposes, the 6.x client using Json.NET internally is an implementation detail and types from a referenced Json.NET assembly are not related nor know how to handle the internalized Json.NET types.

An accurate test case would be

```c#
private static void Main()
{
    var defaultIndex = "default_index";
	var pool = new SingleNodeConnectionPool(new Uri("http://localhost:9200"));
	var settings = new ConnectionSettings(pool, new InMemoryConnection())
		.DefaultIndex(defaultIndex)
		.DisableDirectStreaming()
		.PrettyJson()
		.OnRequestCompleted(callDetails =>
		{
			if (callDetails.RequestBodyInBytes != null)
			{
				Console.WriteLine(
					$"{callDetails.HttpMethod} {callDetails.Uri} \n" +
					$"{Encoding.UTF8.GetString(callDetails.RequestBodyInBytes)}");
			}
			else
			{
				Console.WriteLine($"{callDetails.HttpMethod} {callDetails.Uri}");
			}
		});

	var client = new ElasticClient(settings);

	client.IndexDocument(new Document
	{
		Location = new GeoLocation(45, 45)
	});
}

public class Document
{
	public GeoLocation Location { get; set; }
}
```

which, with 6.8.3, yields

```
POST http://localhost:9200/default_index/document?pretty=true 
{
  "location": {
    "lat": 45.0,
    "lon": 45.0
  }
}
```

Using the Nest.JsonNetSerializer package, which is the same as above, expect for this

```diff
+ var settings = new ConnectionSettings(pool, new InMemoryConnection(), JsonNetSerializer.Default)
- var settings = new ConnectionSettings(pool, new InMemoryConnection())
```

yields the same result.

If you're using the Nest.JsonNetSerializer and a version older than 6.8.2, you may be hitting this issue https://github.com/elastic/elasticsearch-net/issues/3981 which was fixed in 6.8.2. I would recommend that upgrading the client to the latest 6.x version.
